### ISSUE TITLE
creating data-prototype for the form collection and sub form collection

### ISSUE THREAD

**Issue by kritinemkul** [kristijanhusak/laravel-form-builder#110]
I have a form with the form collection . I have used  prototype() method for the dynamic generation of the parent form but the sub form collection inside the same parent form has also to be generated dynamically but the prototype() method for the sub form collection inside parent form collection is not supported here in same view.
eg code is here : 
<div class="collection-container" data-prototype="{{ form_row($form->formA->prototype()) }}">
                    {!! form_row($form->formA) !!}
                </div>
                <div class="collection-container" data-prototype="{{ form_row($form->formB->prototype()) }}">
                    {!! form_row($form->formA->formB) !!}
              </div>
Here formB is the form collection inside the formA. How could I create prototype for the form collection form B that is inside the form collection formA.

**Comment by kristijanhusak** [comment]
That bug was already reported on L4 version https://github.com/kristijanhusak/laravel4-form-builder/issues/8 . I didn't had time to look into it, since it is a bigger issue. Will probably fix it this week, i hope.

**Comment by kristijanhusak** [comment]
@kritinemkul 
Fixed. Currently only available on dev branch, will be stable when i release 1.6 version.

**Comment by kritinemkul** [comment]
Thank you  @kristijanhusak

**Comment by kristijanhusak** [comment]
@kritinemkul 
Not sure what your structure of forms is, but probably something like this should work:
```
<div class="collection-container" data-prototype="{{ form_row($form->formA->prototype() }}">
    {!! form_row($form->formA) !!}
</div>
```

**Comment by kritinemkul** [comment]
@kristijanhusak 
The code you have given above works finely for a collection but if i need the sub form like Form B instead of Form A where From B collection is inside the Form A collection then the code doesnot works. eg is right here,
i have the form structure like;
public function buildForm()
{
       $this
           ->add('form1', 'collection', [
                'type' => 'form',
                'prototype' => true,
                'options' => [
                    'class' => 'path to form',
                    'label' => false,
                ]
            ])
           ->add('form2', 'collection', [
                'type' => 'form',
                'prototype' => true,
                'options' => [
                    'class' => 'path to form',
                    'label' => false,
                ]
            ]);
}

here if i need to duplicate whole this form then the code works like a charm but if i need to duplicate only form 1 then what should i do .

**Comment by kristijanhusak** [comment]
@kritinemkul 
Just use the prototype on the collection. 
```
{!! form_start($form) !!}
<h1>Form1</h1>
<div class="collection-container" data-prototype="{{ form_row($form->form1->prototype()) }}">
  {!! form_row($form->form1) !!}
</div>

<h1>Form2</h1>
<div class="collection-container" data-prototype="{{ form_row($form->form2->prototype()) }}">
  {!! form_row($form->form2) !!}
</div>
{!! form_end($form) !!}
```

**Comment by kristijanhusak** [comment]
@kritinemkul 
Did you manage to set it up?

**Comment by kritinemkul** [comment]
@kristijanhusak 
Thanks for your support . I have tried but it doesnot work dont know what goes wrong and as the solution is just only in dev branch , I have decided to use the setup after you release the stable version. Hope the stable version will release soon.

**Comment by kristijanhusak** [comment]
@kritinemkul 
I released version 1.6 . Do a composer update and check if it works properly.

**Comment by kristijanhusak** [comment]
Closing since there is no negative feedback after the update.
